dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/dataintegrity/AbstractDataIntegrityIntegrationTest.java

```diff
@@ -0,0 +1,241 @@
+        }
+        assertEquals( issueType, details.getIssuesIdType() );
+    }
+
+    public final void DataIntegrityPositiveTestTemplate( String issueType, String check, Integer expectedCount,
+        Integer expectedPercentage, Set<String> expectedDetailsUnits,
+        Set<String> expectedDetailsNames, Set<String> expectedDetailsComments )
+    {
+
+        postSummary( check );
+        JsonDataIntegritySummary summary = GET( "/dataIntegrity/" + check + "/summary" )
+            .content()
+            .as( JsonDataIntegritySummary.class );
+        assertTrue( summary.exists() );
+        assertTrue( summary.isObject() );
+        assertEquals( expectedCount, summary.getCount() );
+        assertEquals( expectedPercentage, summary.getPercentage().intValue() );
+
+        postDetails( check );
+
+        JsonDataIntegrityDetails details = GET( "/dataIntegrity/" + check + "/details" )
+            .content()
+            .as( JsonDataIntegrityDetails.class );
+        assertTrue( details.exists() );
+        assertTrue( details.isObject() );
+        JsonList<JsonDataIntegrityDetails.JsonDataIntegrityIssue> issues = details.getIssues();
+        assertTrue( issues.exists() );
+
+        Set issueUIDs = issues.stream().map( issue -> issue.getId() ).collect( Collectors.toSet() );
+        assertEquals( issueUIDs, expectedDetailsUnits );
+
+        if ( expectedDetailsNames != null )
+        {
+            Set detailsNames = issues.stream().map( issue -> issue.getName() ).collect( Collectors.toSet() );
+            assertEquals( expectedDetailsNames, detailsNames );
+        }
+        if ( expectedDetailsComments != null )
+        {
+            Set detailsComments = issues.stream().map( issue -> issue.getComment() ).collect( Collectors.toSet() );
+            assertEquals( expectedDetailsNames, detailsComments );
+        }
+
+        assertEquals( issueType, details.getIssuesIdType() );
+    }
+
+    public final void DataIntegrityDivideByZeroTestTemplate( String detailsIssueType, String check )
+    {
+        postSummary( check );
+
+        JsonDataIntegritySummary summary = GET( "/dataIntegrity/" + check + "/summary" )
+            .content()
+            .as( JsonDataIntegritySummary.class );
+        assertTrue( summary.exists() );
+        assertTrue( summary.isObject() );
+        assertEquals( 0, summary.getCount() );
+        assertNull( summary.getPercentage() );
+
+        postDetails( check );
+
+        JsonDataIntegrityDetails details = GET( "/dataIntegrity/" + check + "/details" )
+            .content()
+            .as( JsonDataIntegrityDetails.class );
+        assertTrue( details.exists() );
+        assertTrue( details.isObject() );
+        JsonList<JsonDataIntegrityDetails.JsonDataIntegrityIssue> issues = details.getIssues();
+        assertTrue( issues.exists() );
+        assertEquals( 0, issues.size() );
+    }
+
+    public final void DataIntegrityNegativeTestTemplate( String detailsIssueType, String check )
```

Comment:
This looks almost the same as the one above just that it expects a percent 0 instead percent null. 
You can make them one method with an argument for the expected percent:
```java
void assertNoIssues( tring detailsIssueType, String check, boolean percentageIsPresent )
```